extensions/ql-vscode/src/run-queries.ts

```diff
@@ -357,15 +357,15 @@ export class QueryEvaluationInfo {
    * Calls the appropriate CLI command to generate a JSON log summary and parse it 
    * into the appropriate data model for the log visualizer. 
    */
-  parseJsonLogSummary(cliServer: cli.CodeQLCliServer): void {
+  parseJsonLogSummary(queryInfo: LocalQueryInfo, cliServer: cli.CodeQLCliServer): void {
     void cliServer.generateJsonLogSummary(this.evalLogPath, this.jsonEvalLogSummaryPath)
-      .then(() => {         
+      .then(() => {
         // TODO(angelapwen): Stream the file in. 
         fs.readFile(this.jsonEvalLogSummaryPath, (err, buffer) => {
           if (err) {
             throw new Error(`Could not read structured evaluator log summary JSON file at ${this.jsonEvalLogSummaryPath}.`);
           }
-          parseVisualizerData(buffer.toString()); // Eventually this return value will feed into the tree visualizer.
+          queryInfo.evalLogVisualizerData = parseVisualizerData(buffer.toString());
```

Comment:
I ended up moving this method to `query-history.ts`, triggered only when the viewer command is clicked, so that I could avoid passing around all of the data in the `queryInfo` as you mentioned earlier. Now, the JSON summary log for the viewer is generated here (in canary mode) but it isn't read until the command is clicked. 

I think the parsing logic makes more sense in the Query History view anyway as well 😄